static/app/components/searchQueryBuilder/tokens/filterKeyListBox/useRecentSearchFilters.tsx

```diff
@@ -64,7 +65,14 @@ function getFiltersFromRecentSearches(
     .flatMap(search =>
       getFiltersFromQuery({query: search.query, getFieldDefinition, filterKeys})
     )
-    .filter(filter => !filtersInCurrentQuery.includes(filter))
+    .filter(filter => {
+      if (isAggregateFieldOrEquation(filter)) {
```

Comment:
Is this necessary? We probably want to treat aggregates the same as other filter keys. If we support that filter key we should show it, otherwise we should not.

I think part of the reason is that `getFiltersFromQuery` doesn't handle aggregate filters correctly. Try changing that function to return `getKeyName(token.key)` instead of `token.key.text`

Reply:
This works @malwilley , I added a test too.